linera-rpc/src/grpc/client.rs

```diff
@@ -94,43 +94,77 @@ impl GrpcClient {
             }
         }
     }
-}
 
-macro_rules! client_delegate {
-    ($self:ident, $handler:ident, $req:ident) => {{
-        debug!(request = ?$req, "sending gRPC request");
-        let request_inner = $req.try_into().map_err(|_| NodeError::GrpcError {
-            error: "could not convert request to proto".to_string(),
+    async fn delegate<F, FUT, R, S>(
+        &self,
+        f: F,
+        request: impl TryInto<R> + fmt::Debug + Clone,
+        handler: &str,
+    ) -> Result<S, NodeError>
+    where
+        F: Fn(ValidatorNodeClient<transport::Channel>, Request<R>) -> FUT,
+        FUT: Future<Output = Result<tonic::Response<S>, Status>>,
+        R: IntoRequest<R>,
+    {
+        debug!(request = ?request, "sending gRPC request");
+        let mut retry_count = 0;
+        loop {
+            let request_inner = request
+                .clone()
+                .try_into()
+                .map_err(|_| NodeError::GrpcError {
+                    error: "could not convert request to proto".to_string(),
+                })?;
+            match f(self.client.clone(), Request::new(request_inner)).await {
+                Err(s) if Self::is_retryable(&s) && retry_count < self.max_retries => {
+                    let delay = self.retry_delay.saturating_mul(retry_count);
```

Comment:
since you initialise `retry_count = 0`, the first `delay` will be 0, right? I assume we want the first delay to be `self.retry_delay`?

Reply:
I think in most cases it makes sense to retry immediately, at least once?

Reply:
I'd still put a `delay` but (gonna pull my "joker card") I don't know the system yet well enough to be sure.

Reply:
Happy to discuss and possibly change that, but maybe in a later PR? This was already the behavior (for notification streams) before this one.